re/plugins/sink/mainflux.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/emqx/kuiper/common"
+	"github.com/emqx/kuiper/xstream/api"
+	"github.com/mainflux/mainflux/pkg/messaging"
+	"github.com/mainflux/mainflux/pkg/messaging/nats"
+	"github.com/mainflux/senml"
+)
+
+const baseName = "kuiper"
+
+type mainfluxConfig struct {
+	Host     string `json:"host"`
+	Port     string `json:"port"`
+	Channel  string `json:"channel"`
+	Subtopic string `json:"subtopic"`
+}
+
+type mainfluxSink struct {
+	cfg *mainfluxConfig
+	pub nats.Publisher
+}
+
+// Configure sink with properties from rule action definition
+func (ms *mainfluxSink) Configure(props map[string]interface{}) error {
+	cfg := &mainfluxConfig{}
+
+	if err := common.MapToStruct(props, cfg); err != nil {
+		return fmt.Errorf("Read properties %v fail with error: %v", props, err)
+	}
+	if cfg.Host == "" {
+		return fmt.Errorf("Property Host is required.")
+	}
+	if cfg.Port == "" {
+		return fmt.Errorf("Property Port is required.")
+	}
+	if cfg.Channel == "" {
+		return fmt.Errorf("Property Channel is required.")
+	}
+
+	ms.cfg = cfg
+
+	return nil
+}
+
+func (ms *mainfluxSink) Open(ctx api.StreamContext) (err error) {
+	logger := ctx.GetLogger()
+	logger.Debug("Opening mainflux sink")
+
+	addr := fmt.Sprintf("tcp://%s:%s/", ms.cfg.Host, ms.cfg.Port)
+	pub, err := nats.NewPublisher(addr)
+	if err != nil {
+		return fmt.Errorf("Failed to connect to nats at address %s with error: %v", addr, err)
+	}
+	ms.pub = pub
+
+	return
+}
+
+// Collect publishes messages transferred to sink to nats
+func (ms *mainfluxSink) Collect(ctx api.StreamContext, item interface{}) error {
+	logger := ctx.GetLogger()
+	logger.Debugf("mainflux sink receive %v", item)
```

Comment:
we have our logger, should we use it?

Reply:
No. This is a kuiper plugin. This code is run inside kuiper and is not a part of Mainflux. I don't think that Mainflux logger is appropriate here.

Reply:
Is this in completely other container? Then logger is probably already set in the Kuiper init... Let's think about this one, I would not mind setting up Mainflux logger for this. But let's think a bit.